lightning/src/ln/channel.rs

```diff
@@ -3771,6 +3831,64 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 		self.channel_transaction_parameters.channel_type_features = self.channel_type.clone();
 		Ok(())
 	}
+
+	// /// Splice process starting; update state; update capacity, state, reset funding tx
+	// #[cfg(splicing)]
+	// pub(crate) fn splice_start<L: Deref>(&mut self, is_outgoing: bool, logger: &L) where L::Target: Logger {
+	// 	// Set state, by this point handshake is complete
+	// 	self.channel_state = ChannelState::NegotiatingFunding(NegotiatingFundingFlags::OUR_INIT_SENT | NegotiatingFundingFlags::THEIR_INIT_SENT);
+
+	// 	log_info!(logger,
+	// 		"Splicing process started, new channel value {}, outgoing {}, channel_id {}",
+	// 		self.channel_value_satoshis, is_outgoing, self.channel_id,
+	// 	);
+	// }
+
+	/// Get the splice message that can be sent during splice initiation.
+	#[cfg(splicing)]
+	pub fn get_splice_init(&self, our_funding_contribution_satoshis: i64, signer_provider: &SP,
+		funding_feerate_perkw: u32, locktime: u32,
+	) -> msgs::SpliceInit {
+		// At this point we are not committed to the new channel value yet, but the funding pubkey
+		// depends on the channel value, so we create here a new funding pubkey with the new
+		// channel value.
+		// Note that channel_keys_id is supposed NOT to change
+		let funding_pubkey = {
```

Comment:
Is there is really need for funding_pubkey? Needs revisiting.

Reply:
I dropped creating new funding pubkey (as it was re-creating the same value anyways), therefore no need for signer generation.